File: php/src/BasicBuff.php
<?php

declare(strict_types=1);

namespace App;

class BasicBuff implements Buff
{
    private float $soak;
    private float $damage;

    public function __construct(float $soak, float $damage)
    {
        $this->soak = $soak;
        $this->damage = $damage;
    }

    public function soakModifier(): float
    {
        return $this->soak;
    }

    public function damageModifier(): float
    {
        return $this->damage;
    }
}

File: go/basic_buff.go
package codingdojo

type BasicBuff struct {
	soak   float64
	damage float64
}

func MakeBasicBuff(soak float64, damage float64) BasicBuff {
	return BasicBuff{soak, damage}
}

func (b BasicBuff) SoakModifier() float64 {
	return b.soak
}

func (b BasicBuff) DamageModifier() float64 {
	return b.damage
}
